### DISCUSSION TITLE
KPN ITV not working, stuck on 85%

### DISCUSSION THREAD

**Discussion by danielpastoor** [fabianishere/udm-iptv#202] [General]
Hello,

I had my setup working but some how the udm has updated it self to version 1.12.33 but sinds then the iptv are not working.

I have tried to reinstall the udm-iptv with the script of this repo. But it still doesn't work.

Info output:
**udm-iptv-diag:**
WAN Interface: eth8
WAN VLAN: 4 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16  217.166.0.0/16
LAN Interfaces: br0
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
=== Service Logs ===
Nov 20 13:36:14 ubnt udm-iptv[19548]: Device iptv already exists.. Deleting device
Nov 20 13:36:14 ubnt udm-iptv[19548]: RTNETLINK answers: Network is down
Nov 20 13:36:14 ubnt udm-iptv[19548]: Obtaining IP address for VLAN interface
Nov 20 13:36:14 ubnt udm-iptv[19548]: udhcpc (v1.22.1) started
Nov 20 13:36:14 ubnt udm-iptv[19548]: Sending discover...
Nov 20 13:36:14 ubnt udm-iptv[19548]: udhcpc: sendto: Network is down
Nov 20 13:36:14 ubnt udm-iptv[19548]: Read error: Network is down, reopening socket
Nov 20 13:36:17 ubnt udm-iptv[19548]: Sending discover...
Nov 20 13:36:17 ubnt udm-iptv[19548]: udhcpc: sendto: Network is down
Nov 20 13:36:17 ubnt udm-iptv[19548]: Read error: Network is down, reopening socket

**systemctl status udm-iptv:**
● udm-iptv.service - IPTV support for the UniFi Dream Machine
   Loaded: loaded (/lib/systemd/system/udm-iptv.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Sun 2022-11-20 13:37:07 CET; 4s ago
  Process: 21099 ExecStart=/usr/lib/udm-iptv/udm-iptv start (code=exited, status=255/EXCEPTION)
 Main PID: 21099 (code=exited, status=255/EXCEPTION)
      CPU: 150ms

Nov 20 13:37:07 ubnt systemd[1]: udm-iptv.service: Consumed 150ms CPU time.

**zcat /proc/config.gz | grep "MROUTE"**
CONFIG_IP_MROUTE_COMMON=y
CONFIG_IP_MROUTE=y
# CONFIG_IP_MROUTE_MULTIPLE_TABLES is not set
# CONFIG_IPV6_MROUTE is not set

Did I configure something wrong or do I need te reinstall an older version.

**Comment by fabianishere** [comment]
Please remove the `iptv` interface and restart the service:

```bash
ip link del dev iptv
systemctl restart udm-iptv
```

**Reply by danielpastoor** [reply]
Thanks you for replying. 

I'm still getting the same error when i'm looking at the status of the udm-tv service

**Reply by fabianishere** [reply]
Could you share the output of `ip a`?

**Reply by danielpastoor** [reply]
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback d2:21:f9:85:23:59 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: dummy0: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN group default qlen 1000
    link/ether e6:5f:16:22:98:5b brd ff:ff:ff:ff:ff:ff
3: eth9: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 10000
    link/ether d0:21:f9:85:23:62 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d221:f9ff:fe85:2362/64 scope link 
       valid_lft forever preferred_lft forever
4: eth8: <BROADCAST,MULTICAST> mtu 1500 qdisc mq state DOWN group default qlen 1000
    link/ether d0:21:f9:85:23:61 brd ff:ff:ff:ff:ff:ff
5: eth10: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc mq master br0 state DOWN group default qlen 10000
    link/ether d0:21:f9:85:23:63 brd ff:ff:ff:ff:ff:ff
6: switch0: <BROADCAST,MULTICAST,PROMISC,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether d2:21:f9:85:23:5a brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d021:f9ff:fe85:235a/64 scope link 
       valid_lft forever preferred_lft forever
7: gre0@NONE: <NOARP> mtu 1476 qdisc noop state DOWN group default qlen 1000
    link/gre 0.0.0.0 brd 0.0.0.0
8: gretap0@NONE: <BROADCAST,MULTICAST> mtu 1462 qdisc noop state DOWN group default qlen 1000
    link/ether 00:00:00:00:00:00 brd ff:ff:ff:ff:ff:ff
9: erspan0@NONE: <BROADCAST,MULTICAST> mtu 1450 qdisc noop state DOWN group default qlen 1000
    link/ether 00:00:00:00:00:00 brd ff:ff:ff:ff:ff:ff
10: ip_vti0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
    link/ipip 0.0.0.0 brd 0.0.0.0
11: sit0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
    link/sit 0.0.0.0 brd 0.0.0.0
12: br0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d2:21:f9:85:23:5a brd ff:ff:ff:ff:ff:ff
    inet 192.168.1.1/24 scope global br0
       valid_lft forever preferred_lft forever
    inet6 fe80::d021:f9ff:fe85:235a/64 scope link 
       valid_lft forever preferred_lft forever
13: eth0@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d0:21:f9:85:23:59 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d221:f9ff:fe85:2359/64 scope link 
       valid_lft forever preferred_lft forever
14: eth1@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d0:21:f9:85:23:5a brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d221:f9ff:fe85:235a/64 scope link 
       valid_lft forever preferred_lft forever
15: eth2@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d0:21:f9:85:23:5b brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d221:f9ff:fe85:235b/64 scope link 
       valid_lft forever preferred_lft forever
16: eth3@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d0:21:f9:85:23:5c brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d221:f9ff:fe85:235c/64 scope link 
       valid_lft forever preferred_lft forever
17: eth4@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d0:21:f9:85:23:5d brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d221:f9ff:fe85:235d/64 scope link 
       valid_lft forever preferred_lft forever
18: eth5@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d0:21:f9:85:23:5e brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d221:f9ff:fe85:235e/64 scope link 
       valid_lft forever preferred_lft forever
19: eth6@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d0:21:f9:85:23:5f brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d221:f9ff:fe85:235f/64 scope link 
       valid_lft forever preferred_lft forever
20: eth7@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d0:21:f9:85:23:60 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d221:f9ff:fe85:2360/64 scope link 
       valid_lft forever preferred_lft forever
21: ifb0: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN group default qlen 32
    link/ether a6:32:31:4b:0b:9d brd ff:ff:ff:ff:ff:ff
22: ifb1: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN group default qlen 32
    link/ether ba:b0:bf:d4:24:36 brd ff:ff:ff:ff:ff:ff
23: eth10.20@eth10: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue master br20 state LOWERLAYERDOWN group default qlen 1000
    link/ether d0:21:f9:85:23:63 brd ff:ff:ff:ff:ff:ff
24: eth10.30@eth10: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue master br30 state LOWERLAYERDOWN group default qlen 1000
    link/ether d0:21:f9:85:23:63 brd ff:ff:ff:ff:ff:ff
25: eth10.4@eth10: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue master br4 state LOWERLAYERDOWN group default qlen 1000
    link/ether d0:21:f9:85:23:63 brd ff:ff:ff:ff:ff:ff
26: eth10.6@eth10: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue master br6 state LOWERLAYERDOWN group default qlen 1000
    link/ether d0:21:f9:85:23:63 brd ff:ff:ff:ff:ff:ff
27: eth10.7@eth10: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue master br7 state LOWERLAYERDOWN group default qlen 1000
    link/ether d0:21:f9:85:23:63 brd ff:ff:ff:ff:ff:ff
28: eth9.6@eth9: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d0:21:f9:85:23:62 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d221:f9ff:fe85:2362/64 scope link 
       valid_lft forever preferred_lft forever
29: switch0.1@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br0 state UP group default qlen 1000
    link/ether d2:21:f9:85:23:5a brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d021:f9ff:fe85:235a/64 scope link 
       valid_lft forever preferred_lft forever
30: switch0.20@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br20 state UP group default qlen 1000
    link/ether d2:21:f9:85:23:5a brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d021:f9ff:fe85:235a/64 scope link 
       valid_lft forever preferred_lft forever
31: switch0.30@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br30 state UP group default qlen 1000
    link/ether d2:21:f9:85:23:5a brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d021:f9ff:fe85:235a/64 scope link 
       valid_lft forever preferred_lft forever
32: switch0.4@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br4 state UP group default qlen 1000
    link/ether d2:21:f9:85:23:5a brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d021:f9ff:fe85:235a/64 scope link 
       valid_lft forever preferred_lft forever
33: switch0.6@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br6 state UP group default qlen 1000
    link/ether d2:21:f9:85:23:5a brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d021:f9ff:fe85:235a/64 scope link 
       valid_lft forever preferred_lft forever
34: switch0.7@switch0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br7 state UP group default qlen 1000
    link/ether d2:21:f9:85:23:5a brd ff:ff:ff:ff:ff:ff
    inet6 fe80::d021:f9ff:fe85:235a/64 scope link 
       valid_lft forever preferred_lft forever
35: br20: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d0:21:f9:85:23:63 brd ff:ff:ff:ff:ff:ff
    inet 172.16.0.1/24 scope global br20
       valid_lft forever preferred_lft forever
    inet6 fe80::9cf1:5aff:fea4:6060/64 scope link 
       valid_lft forever preferred_lft forever
36: br30: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d0:21:f9:85:23:63 brd ff:ff:ff:ff:ff:ff
    inet 172.16.1.1/24 scope global br30
       valid_lft forever preferred_lft forever
    inet6 fe80::e03d:edff:fee1:db82/64 scope link 
       valid_lft forever preferred_lft forever
37: br4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d0:21:f9:85:23:63 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::3c5e:36ff:fe8b:5306/64 scope link 
       valid_lft forever preferred_lft forever
38: br6: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d0:21:f9:85:23:63 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::84be:4aff:fe23:9f53/64 scope link 
       valid_lft forever preferred_lft forever
39: br7: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether d0:21:f9:85:23:63 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::f4cd:54ff:fe5e:1aa2/64 scope link 
       valid_lft forever preferred_lft forever
50: ppp0: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1492 qdisc pfifo_fast state UNKNOWN group default qlen 3    link/ppp 
    inet 86.--.--.144 peer 195.190.228.33/32 scope global ppp0
       valid_lft forever preferred_lft forever
8326: iptv@eth8: <BROADCAST,MULTICAST,M-DOWN> mtu 1500 qdisc noop state DOWN group default qlen 1000
    link/ether d0:21:f9:85:23:61 brd ff:ff:ff:ff:ff:ff
    
    I replace some numbers to "-" in my public ip before sharing it here

**Reply by fabianishere** [reply]
It seems you are using eth9 (Port 10) for WAN port, but did not update the config for `udm-iptv`.

**Reply by danielpastoor** [reply]
I thought I configured it correctly but I have changed the config and the udm-iptv service has started correctly. 

Big Thanks for this script and for your help.

**Reply by lepopelepope** [reply]
Hi I have the same issue, what is the command to re-configure the ports? I am using eth9 too but cannot acces the config file, at least so it seems.

Thx

**Reply by danielpastoor** [reply]
You can reconfigure it with the following commando:

dpkg-reconfigure -p medium udm-iptv

**Reply by lepopelepope** [reply]
I used that command several times, I do get the GUI but there is no option to change the eth-ports

**Reply by danielpastoor** [reply]
Maybe try to install a newer version of this script. Or are you already on the newest version

**Reply by lepopelepope** [reply]
I am running the latest, maybe a reinstall..

**Reply by lepopelepope** [reply]
I see eth9 = SPF that actually should be eth10. Got it changed to eth9 but still no luck and stuck on 85%..

**Reply by fabianishere** [reply]
eth8 = RJ45 WAN port (Port 9)
eth9 = SFP+ WAN port (Port 10)

**Reply by lepopelepope** [reply]
got it working finally, seems that I need to take out the UDM Flex in between...

**Reply by ARTSRAY** [reply]
For the first time in weeks after installing UDM SE and the Fabian script I had to reinstall the configuration to resolve a 400 error on my KPN mediabox. Maybe an update to the UDM SE triggered the issue. After running the script again it immediately passed 85% on the mediabox and IPTV started working.
